src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -183,12 +183,12 @@ def animal_energy_units(geom):
         # Convert result to dictionary
         columns = [col[0] for col in cursor.description]
         values = cursor.fetchone()  # Only one row since aggregate query
-        population = dict(zip(columns, values))
-
-        livestock_aeu = round(sum(population.get(animal, 0) *
+        population = {k: v or 0.0 for (k, v) in
```

Comment:
Instead of doing this in Python, it may be more efficient to do it inside the query itself. Something like this perhaps, using [COALESCE](https://www.postgresql.org/docs/9.2/static/functions-conditional.html#FUNCTIONS-COALESCE-NVL-IFNULL):

```diff
diff --git a/src/mmw/apps/modeling/mapshed/calcs.py b/src/mmw/apps/modeling/mapshed/calcs.py
index 659c927..10253f2 100644
--- a/src/mmw/apps/modeling/mapshed/calcs.py
+++ b/src/mmw/apps/modeling/mapshed/calcs.py
@@ -161,19 +161,19 @@ def animal_energy_units(geom):
                                   ST_SetSRID(ST_GeomFromText(%s),
                                              4326))
           ), clipped_counties_with_area AS (
-              SELECT ST_Area(geom_clipped) / ST_Area(geom) AS clip_percent,
+              SELECT ST_Area(geom_clipped) / ST_Area(geom) AS clip_pct,
                      clipped_counties.*
               FROM clipped_counties
           )
-          SELECT SUM(beef_ha * ag_ha * clip_percent) AS beef_cows,
-                 SUM(broiler_ha * ag_ha * clip_percent) AS broilers,
-                 SUM(dairy_ha * ag_ha * clip_percent) AS dairy_cows,
-                 SUM(goat_ha * ag_ha * clip_percent) +
-                 SUM(sheep_ha * ag_ha * clip_percent) AS sheep,
-                 SUM(hog_ha * ag_ha * clip_percent) AS hogs,
-                 SUM(horse_ha * ag_ha * clip_percent) AS horses,
-                 SUM(layer_ha * ag_ha * clip_percent) AS layers,
-                 SUM(turkey_ha * ag_ha * clip_percent) AS turkeys
+          SELECT COALESCE(SUM(beef_ha * ag_ha * clip_pct), 0.0) AS beef_cows,
+                 COALESCE(SUM(broiler_ha * ag_ha * clip_pct), 0.0) AS broilers,
+                 COALESCE(SUM(dairy_ha * ag_ha * clip_pct), 0.0) AS dairy_cows,
+                 COALESCE(SUM(goat_ha * ag_ha * clip_pct), 0.0) +
+                 COALESCE(SUM(sheep_ha * ag_ha * clip_pct), 0.0) AS sheep,
+                 COALESCE(SUM(hog_ha * ag_ha * clip_pct), 0.0) AS hogs,
+                 COALESCE(SUM(horse_ha * ag_ha * clip_pct), 0.0) AS horses,
+                 COALESCE(SUM(layer_ha * ag_ha * clip_pct), 0.0) AS layers,
+                 COALESCE(SUM(turkey_ha * ag_ha * clip_pct), 0.0) AS turkeys
           FROM clipped_counties_with_area;
           '''
 
```

I had to shorten `clip_percent` to `clip_pct` to get it all to fit within 80 characters.